### PR TITLE
records: update index settings for accent search

### PR DESCRIPTION
**Pull request by sakshamarora1**
closes: https://github.com/inveniosoftware/invenio-app-rdm/issues/2746
closes: https://github.com/inveniosoftware/invenio-app-rdm/issues/2766

docs for reference:
https://opensearch.org/docs/latest/analyzers/#built-in-analyzers
https://opensearch.org/docs/latest/analyzers/token-filters/index/

`strip_special_chars` is a custom char_filter which uses pattern_replace (https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-charfilters.html)

### CODE REVIEW

**Comment by carlinmack** [comment]
I've only checked os-v2 but it seems like there's an extra level of indentation on the whole file after the first line

**Comment by carlinmack** [comment]
It also seems like the entire `mappings.properties.parent.properties.communities.properties.entries` path has been deleted, around 200 lines, is this on purpose?

File: invenio_rdm_records/records/jsonschemas/records/record-v7.0.0.json [hunk 1]
```diff
@@ -0,0 +1,460 @@
+{
```
Comment (jrcastro2) [inline invenio_rdm_records/records/jsonschemas/records/record-v7.0.0.json:1]: Why do we need this one? it looks like it's the same content as v6, unless I am missing something, see here comparaison: https://www.diffchecker.com/FZPE8bDa/

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json [hunk 2]
```diff
@@ -0,0 +1,1480 @@
...
+      "metadata.subjects.subject",
+      "metadata.version"
+    ],
+    "analysis": {
+      "char_filter": {
+        "strip_special_chars": {
+          "type": "pattern_replace",
+          "pattern": "[\\p{Punct}\\p{S}]",
+          "replacement": ""
+        }
```
Comment (jrcastro2) [inline invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:1461]: minor: I see that we have also removed special chars, by doing so we are affecting equations, let's say we have the following input in the title "a/b = c*d - e", with this changes if I want to search for it I would need to write "ab  cd  e", which doesn't seem very intuitive for the user. I would leave this one out from RDM, unless we can match both cases, WDYT?

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json [hunk 3]
```diff
@@ -0,0 +1,1480 @@
...
+            "type": "text",
+            "fields": {
+              "keyword": {
+                "type": "keyword",
+                "ignore_above": 256,
+                "normalizer": "custom_normalizer"
+              }
+            },
+            "analyzer": "custom_analyzer",
+            "search_analyzer": "custom_analyzer"
```
Comment (jrcastro2) [inline invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:1111]: Comment: Should this also be applied to other fields such as additional titles, ... ?

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json (hunk 2 above)
Comment (sakshamarora1) [reply invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:1479]: Added more fields!

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json (hunk 3 above)
Comment (sakshamarora1) [reply invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json]: Added more fields!

File: invenio_rdm_records/records/jsonschemas/records/record-v7.0.0.json (hunk 1 above)
Comment (sakshamarora1) [reply invenio_rdm_records/records/jsonschemas/records/record-v7.0.0.json:1]: Yes but do we not need to add the same schema for v7 if we are adding v7 mappings? is it different from ILS?

**Review by jrcastro2 — APPROVED** [review]
LGTM! 🚀 

Maybe in the future we want to apply these to more fields but it's a good start!

File: invenio_rdm_records/records/mappings/os-v1/rdmrecords/records/record-v7.0.0.json [hunk 4]
```diff
@@ -0,0 +1,1491 @@
...
+              }
+            }
+          }
+        }
+      }
+    ],
+    "dynamic": "strict",
+    "date_detection": false,
+    "numeric_detection": false,
+    "properties": {
```
Comment (kpsherva) [inline invenio_rdm_records/records/mappings/os-v1/rdmrecords/records/record-v7.0.0.json:76]: for which fields did you apply the new analysers ?

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json (hunk 2 above)
Comment (kpsherva) [reply invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:1479]: was it addressed? could someone confirm?

File: invenio_rdm_records/records/mappings/os-v1/rdmrecords/records/record-v7.0.0.json (hunk 4 above)
Comment (sakshamarora1) [reply invenio_rdm_records/records/mappings/os-v1/rdmrecords/records/record-v7.0.0.json:76]: It's the titles, additional titles, description, and name sub fields of creators, contributors and person_or_org
Here is the diff for the record index:
```diff
@@ -613,7 +613,9 @@
                     "type": "keyword"
                   },
                   "name": {
-                    "type": "text"
+                    "type": "text",
+                    "analyzer": "accent_analyzer",
+                    "search_analyzer": "accent_analyzer"
                   },
                   "identifiers": {
                     "properties": {
@@ -682,7 +684,9 @@
                     "type": "keyword"
                   },
                   "name": {
-                    "type": "text"
+                    "type": "text",
+                    "analyzer": "accent_analyzer",
+                    "search_analyzer": "accent_analyzer"
                   },
                   "identifiers": {
                     "properties": {
@@ -715,7 +719,9 @@
                     }
                   },
                   "name": {
-                    "type": "text"
+                    "type": "text",
+                    "analyzer": "accent_analyzer",
+                    "search_analyzer": "accent_analyzer"
                   },
                   "type": {
                     "type": "keyword"
@@ -768,7 +774,9 @@
             }
           },
           "description": {
-            "type": "text"
+            "type": "text",
+            "analyzer": "accent_analyzer",
+            "search_analyzer": "accent_analyzer"
           },
           "additional_descriptions": {
             "properties": {
@@ -1100,10 +1108,15 @@
           },
           "title": {
             "type": "text",
+            "analyzer": "accent_analyzer",
+            "search_analyzer": "accent_analyzer",
             "fields": {
               "keyword": {
                 "type": "keyword",
                 "ignore_above": 256
+              },
+              "original": {
+                "type": "text"
               }
             }
           },
@@ -1125,7 +1138,14 @@
                 }
               },
               "title": {
-                "type": "text"
+                "type": "text",
+                "analyzer": "accent_analyzer",
+                "search_analyzer": "accent_analyzer",
+                "fields": {
+                  "original": {
+                    "type": "text"
+                  }
+                }
               },
               "type": {
                 "type": "object",
```

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json (hunk 2 above)
Comment (sakshamarora1) [reply invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:1479]: Sorry I missed replying
Yes, this has been addressed by adding an `original` title sub field which doesn't use any analyzers.
So we are not matching both cases.

File: tests/resources/test_resources_search.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+
+def test_advance_record_search(
+    client,
+    headers,
+    record_community,
+    minimal_record,
+    uploader,
+):
+    """Test search record's communities."""
```
Comment (anikachurilova) [inline tests/resources/test_resources_search.py:20]: Could you please also add a test with punctuation? For example, searching by just `title` and getting no results, then trying to search by `title.original`?

**Review by zzacharo — APPROVED** [review]

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json [hunk 6]
```diff
@@ -0,0 +1,1491 @@
+{
+  "settings": {
+    "index.query.default_field": [
```
Comment (ntarocco) [inline invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:3]: While reviewing this, me and @kpsherva were wondering if we are all ok with restricting the common search (without specifying the field(s)) to these fields. A quick discussion would be nice.

In any case, it looks like that the `text` fields have been added here, which makes sense. However these are missing:
```
dates.description <--- probably not needed
additional_descriptions.description
references.reference
additional_titles.title
```

Given that `pids` is not added, if we search for a DOI, will the record be found?

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json [hunk 7]
```diff
@@ -0,0 +1,1491 @@
...
+      "id",
+      "metadata.title",
+      "metadata.title.original",
```
Comment (ntarocco) [inline invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:6]: the `title.original` is not part of the mapping, but looks like coming from the `dynamic` section, correct?

File: tests/resources/test_resources_search.py (hunk 5 above)
Comment (ntarocco) [reply tests/resources/test_resources_search.py:20]: Looks like done :)

File: invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json (hunk 7 above)
Comment (sakshamarora1) [reply invenio_rdm_records/records/mappings/os-v2/rdmrecords/records/record-v7.0.0.json:6]: It is part of the mapping
[See line 1164](https://github.com/inveniosoftware/invenio-rdm-records/pull/1774/files#diff-40a24708a1fed90c5895e1320701bb212c6230ad4bf6d9846a67e973cc5e9b23R1164)

**Pull request merged**